From c6d1de4226864e9874c4b7541b31ea04a96f106c Mon Sep 17 00:00:00 2001
From: Robin Kauffman <robink@creosotehill.org>
Date: Fri, 16 Aug 2019 17:11:36 -0700
Subject: [PATCH] Make documentation configurable w/ meson

Documentation generation was initially set up to only be configurable
with autotools.  (Hopefully) fix that.
---
 meson.build       | 4 ++--
 meson_options.txt | 4 ++++
 2 files changed, 6 insertions(+), 2 deletions(-)

diff --git a/meson.build b/meson.build
index 908811b..41c0906 100644
--- a/meson.build
+++ b/meson.build
@@ -452,8 +452,8 @@ install_man(configure_file(input: 'tools/libwacom-list-local-devices.man',
 			   copy: true))
 
 ############### docs ###########################
-doxygen = find_program('doxygen', required: false)
-if doxygen.found()
+if get_option('documentation')
+	doxygen = find_program('doxygen', required: true)
 	src_doxygen = [
 		join_paths(dir_src, 'libwacom.h'),
 	]
diff --git a/meson_options.txt b/meson_options.txt
index 1d75434..f59db4c 100644
--- a/meson_options.txt
+++ b/meson_options.txt
@@ -6,4 +6,8 @@ option('tests',
        type: 'boolean',
        value: true,
        description: 'Build the tests [default=true]')
+option('documentation',
+	type: 'boolean',
+	value: 'true',
+	description: 'Build the documentation')
 
-- 
2.20.1

